### ISSUE TITLE
Infinite loop in `_add_lines_to_spectrum`

### ISSUE THREAD

**Issue by maxbeegee** [trident-project/trident#104]
In the function `_add_lines_to_spectrum` in `absorption_spectrum.py`  the variable `window_width_in_bins`
until the following condition is fulfilled:

```
                    # If tau has not dropped below min tau threshold by the
                    # edges (ie the wings), then widen the wavelength
                    # window and repeat process.
                    if (vtau[0] < min_tau and vtau[-1] < min_tau):
                        break
                    window_width_in_bins *= 2

```

However, in some cases this is never fulfilled causing the memory to fill up quickly.
I've uploaded one example [here]( https://www.dropbox.com/s/nidyhyv4c9lson9/ray.h5?dl=1) and this is a MWE illustrating the problem

```
import yt
import trident

ds = yt.load("ray.h5")
line_list = ['H']

sg = trident.SpectrumGenerator(lambda_min = 8480.,
                               lambda_max = 8520.,
                               dlambda = 1e-2)


# changing min_tau to > ~1e-2 resolves the issue
sg.make_spectrum(ds, lines=line_list, min_tau = 1e-3)
```

As stated above, increasing `min_tau` resolves the issue with the cost that other spectra might become less accurate.
Another possibility is to add something like
```
if window_width_in_bins > 1e5:
    break
```
in the loop mentioned above. But this is rather unelegant...

**Comment by zhafen** [comment]
I can confirm what @maxbeegee found, and add that this appears to be a common problem, at least in the cosmological zoom-in simulations I'm working with. It's especially common when passing through highly dense regions, e.g. galaxies.

**Comment by peeples** [comment]
I have also found this behavior, especially for DLA sightlines.

**Comment by brittonsmith** [comment]
@maxbeegee, @zhafen, @peeples there were a few separate issues here regarding array sizes growing too large and missing criteria for considering a given absorption feature successfully deposited. I think I've covered all of these in PR #105. Please have a look and approve/comment on the PR.

**Comment by chummels** [comment]
This issue was resolved by PR #105 . Closing.
